pulsar-client-cpp/lib/ConsumerConfiguration.cc

```diff
@@ -214,6 +214,19 @@ ConsumerConfiguration& ConsumerConfiguration::setProperties(
     return *this;
 }
 
+std::map<std::string, std::string>& ConsumerConfiguration::getSubscriptionProperties() const {
+    return impl_->subscriptionProperties;
+}
+
+ConsumerConfiguration& ConsumerConfiguration::setSubscriptionProperties(
+    const std::map<std::string, std::string>& subscriptionProperties) {
+    for (const auto& subscriptionProperty : subscriptionProperties) {
+        impl_->subscriptionProperties.emplace(
+            std::make_pair(subscriptionProperty.first, subscriptionProperty.second));
```

Comment:
There is no need to emplace a `std::pair`. See following example:

```c++
std::map<std::string, std::string> m;
m.emplace("key", "value");
```

Reply:
ok, it has been revised, thanks for your attention.